flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/typeutils/SqlTimestampTypeInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.typeutils;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.table.dataformat.SqlTimestamp;
+
+import java.util.Objects;
+
+/**
+ * TypeInformation of {@link SqlTimestamp}.
+ */
+public class SqlTimestampTypeInfo extends TypeInformation<SqlTimestamp> {
+
+	private final int precision;
+
+	protected SqlTimestampTypeInfo(int precision) {
```

Comment:
Why protected?

Reply:
Forgot to change it as public when removing the static of function:
`public static SqlTimestampTypeInfo of(int precision)`